### DISCUSSION TITLE
Redirect unauthenticated requests to login-page

### DISCUSSION THREAD

**Discussion by magnusskare** [pocketbase/pocketbase#4479] [Q&A]
I'm probably missing something obvious, but I can't get this to work properly. I'm trying to create my own main.pb.js file, going by the example [in this question](https://github.com/pocketbase/pocketbase/discussions/4420), but while I can get the redirect to work, I'm not seeing any obvious ways to get the current path name.

I would like to redirect all unauthenticated traffic to /login, regardless of route except /login itself. Would this be possible?

From /login, they authenticate with OAuth2. Would a redirect like this mess with this, or would there be better ways to go about doing this?

```
function redirectIfNotAuthenticated(next) {
  return (c) => {
    const user = c.get("authRecord");
    const isLoginPage = false; // Need something here to check if we're already on the login page.
    
    if (!user && !isLoginPage) {
      c.redirect(307, "https://***.pockethost.io/auth")
    }

    return next(c) // proceed with the request chain
  }
}

routerUse(redirectIfNotAuthenticated)
```

**Comment by ganigeorgiev** [comment]
You can try with `c.Path()` if you want to get the router path string or `c.request().url` if you want the full requested url.

```js
console.log("Path:", c.path())
console.log("URL:", c.request().url)
```

You can also link at the top of your `*.pb.js` file the ambient TS declarations to assist you with autocomplete and documentation on hover:

```js
/// <reference path="../pb_data/types.d.ts" />
```
